huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_dataservice_message_approve_test.go

```diff
@@ -0,0 +1,148 @@
+package dataarts
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDataServiceMessageApprove_basic(t *testing.T) {
+	var (
+		name = acceptance.RandomAccResourceName()
+
+		all = "data.huaweicloud_dataarts_dataservice_messages.test"
+		dc  = acceptance.InitDataSourceCheck(all)
+	)
+
+	// lintignore:AT001
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckDataArtsWorkSpaceID(t)
+			acceptance.TestAccPreCheckDataArtsReviewerName(t)
+			acceptance.TestAccPreCheckDataArtsRelatedDliQueueName(t)
+		},
+
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				// Authorize the APP to access the API (do not need to approve).
+				Config: testAccDataServiceMessageApprove_basic_step1(name),
+			},
+			{
+				// Cancel the API access permission of the APP, then generate an approve message and inform the reviewer.
+				Config: testAccDataServiceMessageApprove_basic_step2(name),
```

Comment:
It seems that step2 has no meaning because step3 contains the test script of step2..